sdk/servicebus/azure-messaging-servicebus-track2-perf/src/main/java/com/azure/messaging/servicebus/perf/BatchSendTest.java

```diff
@@ -0,0 +1,76 @@
+package com.azure.messaging.servicebus.perf;
+
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusMessage;
+import com.azure.messaging.servicebus.ServiceBusSenderAsyncClient;
+import com.azure.messaging.servicebus.ServiceBusSenderClient;
+import com.azure.perf.test.core.BatchPerfTest;
+import com.azure.perf.test.core.TestDataCreationHelper;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+public class BatchSendTest extends BatchPerfTest<ServiceBusStressOptions> {
+    private static final ClientLogger LOGGER = new ClientLogger(BatchSendTest.class);
+
+    private static final String AZURE_SERVICE_BUS_CONNECTION_STRING = "AZURE_SERVICE_BUS_CONNECTION_STRING";
+    private static final String AZURE_SERVICEBUS_QUEUE_NAME = "AZURE_SERVICEBUS_QUEUE_NAME";
+
+    private final ServiceBusSenderClient sender;
+    private final ServiceBusSenderAsyncClient senderAsync;
+
+    /**
+     * Creates an instance of Batch performance test.
+     *
+     * @param options the options configured for the test.
+     * @throws IllegalStateException if SSL context cannot be created.
+     */
+    public BatchSendTest(ServiceBusStressOptions options) {
+        super(options);
+
+        String connectionString = System.getenv(AZURE_SERVICE_BUS_CONNECTION_STRING);
+        if (CoreUtils.isNullOrEmpty(connectionString)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                String.format("Environment variable %s must be set", AZURE_SERVICE_BUS_CONNECTION_STRING)));
+        }
+
+        ServiceBusClientBuilder builder = new ServiceBusClientBuilder().connectionString(connectionString);
+        String queueName = System.getenv(AZURE_SERVICEBUS_QUEUE_NAME);
+        if (CoreUtils.isNullOrEmpty(queueName)) {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                String.format("Environment variable %s must be set", AZURE_SERVICEBUS_QUEUE_NAME)));
+        }
+        sender = builder.sender().queueName(queueName).buildClient();
```

Comment:
This whole setup logic can be moved up to an abstract ServiceTest class, so other tests can reuse it.
Have BatchSendTest extend ServiceTest

Reply:
Yeah, when I do this code, I'm considering to create an abstract class, and I found the ServiceTest, but the ServiceTest extend PerfStressTest, and in the gist the BatchSend Test extend BatchPerfTest, so should I create a ServiceBatchTest and move the setup logic (create client) to the new class?